### ISSUE TITLE
Email Address Validation Issue

### ISSUE THREAD

**Issue by teamcoltra** [openpgpjs/openpgpjs#1815]
I'm not sure if this is a bug or a feature request or discussion but in utils:
```
  isEmailAddress: function(data) {
    if (!util.isString(data)) {
      return false;
    }
    const re = /^[^\p{C}\p{Z}@<>\\]+@[^\p{C}\p{Z}@<>\\]+[^\p{C}\p{Z}\p{P}]$/u;
    return re.test(data);
  },
```

It's said that this validation doesn't go far enough but it's actually more restrictive than RFC 5322 but I would argue even more importantly from a compatibility standpoint it isn't in line with GPG which would allow "this-isn't-an-email" (or at least mine does). I was  trying to find any documentation in a standards and RFC 9580:

> [5.11. ](https://www.rfc-editor.org/rfc/rfc9580#section-5.11)[User ID Packet (Type ID 13)](https://www.rfc-editor.org/rfc/rfc9580#name-user-id-packet-type-id-13)
> A User ID packet consists of UTF-8 text that is intended to represent the name and email address of the keyholder. By convention, it includes a mail name-addr as described in [[RFC2822](https://www.rfc-editor.org/rfc/rfc9580#RFC2822)], but there are no restrictions on its content. The packet length in the header specifies the length of the User ID.

Basically, I would advocate a position that email validation doesn't happen or at least is a flag. I get the front-end person wanting to ensure that the user is aware that  their email address may not be valid so they can double check it, but also I don't think that should be part of the back-end work no more than you don't check to make sure that the password is 16+ characters long with a mix of symbols, cases, etc. 

I would call this a bug as it goes outside of the GPG spec.

**Comment by twiss** [comment]
Hi :wave: It's certainly debatable how strict the email validation should be, but I wouldn't call this a bug since the behavior is intentional, although it's true that it's not strictly required by RFC 9580.

However, it's also already possible to include anything you like in the User ID, as follows:
```js
await openpgp.generateKey({ userIDs: [{ name: `this-isn't-an-email` }] });
```
so, I don't think we specifically need an additional flag for that.
